packages/flutter/lib/src/widgets/dismissable.dart

```diff
@@ -340,17 +378,29 @@ class _DismissableState extends State<Dismissable> with TickerProviderStateMixin
       );
     }
 
-    Widget backgroundAndChild = new SlideTransition(
+    Widget content = new SlideTransition(
+      key: new Key('foreground'),
       position: _moveAnimation,
       child: config.child
     );
+
     if (background != null) {
-      backgroundAndChild = new Stack(
-        children: <Widget>[
-          new Positioned(left: 0.0, top: 0.0, bottom: 0.0, right: 0.0, child: background),
-          new Viewport(child: backgroundAndChild)
-        ]
-      );
+      List<Widget> children = <Widget>[];
+
+      if (!_moveAnimation.isDismissed) {
```

Comment:
Ah, I see it now. That makes sense.

Should we put a global key around the background when we insert it? Or should we rely on the caller to do so if they have state in there that they want to preserve across the various ways we build this?

If you don't want to put a global key in this build function to do that, we should document that unless they use a global key in their background, it could get its state reset, and we should suggest putting the same global key on background and secondaryBackground too if they are structurally similar and should have state kept across them.